src/main/java/slimeknights/tconstruct/library/registration/object/RecipeSerializerObject.java

```diff
@@ -0,0 +1,20 @@
+package slimeknights.tconstruct.library.registration.object;
+
+import net.minecraft.item.crafting.IRecipe;
+import net.minecraft.item.crafting.IRecipeSerializer;
+import net.minecraft.util.ResourceLocation;
+
+import java.util.function.Supplier;
+
+public class RecipeSerializerObject<RECIPE extends IRecipeSerializer<?>> implements Supplier<RECIPE> {
```

Comment:
Delete this, `RegistryObject` can already do everything this wrapper can. I only add the wrappers to provide new useful methods, just for the sake of having them.